File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

       int res = 0;
       int curr = 0;
       for (int num : nums) {
        if (!set.contains(num - 1)) {
            int temp = num;
            curr = 1;
            while (set.contains(temp + 1)) {
                curr++;
                temp++;
            }
          }
          res = Math.max(curr, res);
       }
       return res;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        res = 0

        for num in numSet:
            # check if its start of sequence
            # doesn't have left neighbor
            if (num - 1) not in numSet:
                curr_len = 0
                while (num + curr_len) in numSet:
                    curr_len += 1
                
                res = max(res, curr_len)

        return res